File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        while l < r:
            add = numbers[r] + numbers[l]
            if add > target:
                r -= 1
            elif add < target:
                l += 1
            else:
                return [l+1, r+1]
        return []
            

File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> numbers = new HashSet<>();

        for (int num : nums) {
            if (numbers.contains(num)) {
                return true;
            }
            else {
                numbers.add(num);
            }
        }
        return false;
    }
}